### PR TITLE
feat: Add file input and text area to GridForm

### PR DESCRIPTION
**Pull request by mikeboyle**
## [RRT-89] Add file input and text area to GridForm
Adds file input and text area powers to GridForm.
Screen Shot 2020-04-08 at 6 05 47 PM

This required also changing our gamut `<TextArea />` component to use `React.forwardRef` (same as `<Input />`) as well as some styling changes specific to the file input. We are using the `text-indent` attribute for gamut inputs, but this causes the `Choose File` button to cover up the file name in the input.

There are also some changes to types for the select input as well to make it work with more use cases.

### CODE REVIEW

File: packages/gamut-templates/src/GridForm/GridFormInputGroup/GridFormFileInput/index.tsx [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+
+export const GridFormFileInput: React.FC<GridFormFileInputProps> = ({
+  className,
+  field,
+  register,
+  setValue,
+}) => {
+  const onChange = (event: any) => {
+    setValue(event.target.files);
+  };
```
Comment (JoshuaKGoldberg) [inline packages/gamut-templates/src/GridForm/GridFormInputGroup/GridFormFileInput/index.tsx:22]: In my testing, the submit action still worked if `onChange` was omitted altogether. Can we do that for brevity's sake?

If not, let's either make this inline in the `onChange={...}` or give `event` a type? _(also for the other ones)_

Comment (mikeboyle) [reply packages/gamut-templates/src/GridForm/GridFormInputGroup/GridFormFileInput/index.tsx:22]: Losing the onChange seems kind of drastic and setting the value seems like an important part of a react form, but if you're confident in that I can ditch it

Comment (JoshuaKGoldberg) [reply packages/gamut-templates/src/GridForm/GridFormInputGroup/GridFormFileInput/index.tsx:22]: All we care about is that the values sent to `onSubmit` include the value. If that's good, we're good!

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/brand-components@0.8.16-alpha.d719e6.0<br/>@codecademy/gamut-templates@2.7.1-alpha.d719e6.0<br/>@codecademy/gamut@2.18.1-alpha.d719e6.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://5e8f40836f27e23870320569--codecademy-styleguide.netlify.com>https://5e8f40836f27e23870320569--codecademy-styleguide.netlify.com</a></p><p><a href=https://app.netlify.com/sites/codecademy-styleguide/deploys/5e8f40836f27e23870320569>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/brand-components@0.8.16-alpha.a6c311.0<br/>@codecademy/gamut-templates@2.7.1-alpha.a6c311.0<br/>@codecademy/gamut@2.18.1-alpha.a6c311.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://5e8f4ccc9b874e02123b5ab7--codecademy-styleguide.netlify.com>https://5e8f4ccc9b874e02123b5ab7--codecademy-styleguide.netlify.com</a></p><p><a href=https://app.netlify.com/sites/codecademy-styleguide/deploys/5e8f4ccc9b874e02123b5ab7>Deploy Logs</a></p>

File: packages/gamut-templates/src/GridForm/GridFormInputGroup/GridFormFileInput/index.tsx (hunk 1 above)
Comment (JoshuaKGoldberg) [reply packages/gamut-templates/src/GridForm/GridFormInputGroup/GridFormFileInput/index.tsx]: https://github.com/Codecademy/client-modules/pull/712

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/brand-components@0.8.16-alpha.880487.0<br/>@codecademy/gamut@3.0.0-alpha.880487.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://5e8f8e472229b90178d7a1a1--codecademy-styleguide.netlify.com>https://5e8f8e472229b90178d7a1a1--codecademy-styleguide.netlify.com</a></p><p><a href=https://app.netlify.com/sites/codecademy-styleguide/deploys/5e8f8e472229b90178d7a1a1>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/brand-components@0.8.16-alpha.2e845d.0<br/>@codecademy/gamut@3.0.0-alpha.2e845d.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://5e8fd48a84ac0d4a9398dba1--codecademy-styleguide.netlify.com>https://5e8fd48a84ac0d4a9398dba1--codecademy-styleguide.netlify.com</a></p><p><a href=https://app.netlify.com/sites/codecademy-styleguide/deploys/5e8fd48a84ac0d4a9398dba1>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/brand-components@0.8.17-alpha.c7ba8f.0<br/>@codecademy/gamut@3.0.1-alpha.c7ba8f.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://5e8fd765adf82e44dd121db2--codecademy-styleguide.netlify.com>https://5e8fd765adf82e44dd121db2--codecademy-styleguide.netlify.com</a></p><p><a href=https://app.netlify.com/sites/codecademy-styleguide/deploys/5e8fd765adf82e44dd121db2>Deploy Logs</a></p>

File: packages/gamut/src/Form/Input.tsx [hunk 2]
```diff
@@ -16,6 +16,9 @@ export const Input = React.forwardRef<HTMLInputElement, InputProps>(
   ({ error, htmlFor, className, ...rest }, ref) => {
     const classNames = cx(
       s.Input,
+      {
+        [s.fileInput]: rest.type === 'file',
+      },
```
Comment (JoshuaKGoldberg) [inline packages/gamut/src/Form/Input.tsx:21]: It'd be cleaner to use a `.Input[type=file]` in CSS.

**Pull request merged**